sdk/core/azure-core/src/http/raw_response.cpp

```diff
@@ -26,7 +26,7 @@ void RawResponse::SetHeader(uint8_t const* const first, uint8_t const* const las
 
   if (end == last)
   {
-    throw InvalidHeaderException("Invalid header. No delimiter ':' found.");
+    throw std::invalid_argument("Invalid header. No delimiter ':' found.");
```

Comment:
In this case we want to throw `TrasnportException` and not invalid arg.  RawResponse comes only from a transport request.

Reply:
`RawResponse` is a public type in a public header.
`SetHeader` is a public API on that type that accept some parameters.
This check is validating those inputs.